src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -796,7 +800,7 @@ public function isDeprecated()
      */
     public function getDeprecationMessage($id)
     {
-        return str_replace('%service_id%', $id, $this->deprecationTemplate);
+        return null !== $this->deprecationTemplate ? str_replace('%service_id%', $id, $this->deprecationTemplate) : null;
```

Comment:
what's the point then of setting `deprecationMessage` to `null` by default, if we have another property setting a default deprecated message that will be interpreted anyway ?